src/distgen_internal.cpp

```diff
@@ -79,6 +83,36 @@ void addDist(u64 size) {
 	distsUsed++;
 }
 
+static void *thread_func(void *arg) {
```

Comment:
Can you rename that function so it name related to what it actually does? :smile: Maybe something like `init_memory_per_thread`?